src/table/components/head/HeadCellMenu.tsx

```diff
@@ -5,19 +5,35 @@ import Search from '@qlik-trial/sprout/icons/react/Search';
 
 import { useContextSelector, TableContext } from '../../context';
 import { HeadCellMenuProps, MenuItemGroup } from '../../types';
-import { StyledMenuIconButton, NebulaListBox } from './styles';
-import { ListBoxWrapper, ListBoxWrapperRenderProps } from './ListBoxWrapper';
+import { StyledMenuIconButton } from './styles';
 import MenuItems from './MenuItems';
 
-export default function HeadCellMenu({ columnIndex, isDimension, tabIndex }: HeadCellMenuProps) {
-  const { translator } = useContextSelector(TableContext, (value) => value.baseProps);
+export default function HeadCellMenu({ column, tabIndex }: HeadCellMenuProps) {
+  const { translator, embed } = useContextSelector(TableContext, (value) => value.baseProps);
   const [openMenuDropdown, setOpenMenuDropdown] = useState(false);
   const [openListboxDropdown, setOpenListboxDropdown] = useState(false);
-  const anchorRef = useRef<HTMLButtonElement>(null);
+  const anchorRef = useRef<HTMLDivElement>(null);
+  const listboxRef = useRef<HTMLDivElement>(null);
+  const showSearchMenuItem = column.isDim && !column.isMasterItem;
+
+  const embedListbox = () => {
+    // @ts-ignore TODO: no types for `__DO_NOT_USE__`, it will improve when it becomes stable
+    // eslint-disable-next-line
+    embed.__DO_NOT_USE__.popover(listboxRef.current, column.fieldId, {
+      anchorOrigin: { vertical: 'bottom', horizontal: 'left' },
+      transformOrigin: { vertical: 'top', horizontal: 'left' },
+    });
+
+    // @ts-ignore TODO: no types for `__DO_NOT_USE__`, it will improve when it becomes stable
+    // eslint-disable-next-line
```

Comment:
is this needed? Looks like it was copy pasted from above, at least the comment isn't really correct

Reply:
will update it asap